rfcs/react-components/convergence/static-classes-on-slots.md

```diff
@@ -0,0 +1,92 @@
+# RFC: Conventions for static class names on slots
+
+Contributors: @behowell
+
+History:
+
+- [PR #21206](https://github.com/microsoft/fluentui/pull/21206) - Initial submission
+
+## Summary
+
+This RFC proposes adding a static className to every slot, using the naming format `"fui-{ComponentName}__{slotName}"`.
+
+## Problem statement
+
+It can be useful to have a static class name on the slot of a component, including by users of our library if they want to write static styles targeting a slot.
+
+We currently have a static class name `"fui-{ComponentName}"` (e.g. `"fui-Button"`) added to the root slot of every component, as described in issue [#19937](https://github.com/microsoft/fluentui/issues/19937). However, we don't currently have any formal specification of the format of the classNames for other slots, nor any conformance tests ensuring that they are added to every slot.
+
+Some of our components already have static classNames on slots (e.g. Checkbox's `indicator` slot). However, this patchwork addition of static classes makes it unpredictable for users of our library.
+
+One example internal usage would be to write a style such as this, which changes the Checkbox `indicator` slot's backgroundColor when hovering over the `root`:
+
+```ts
+const useStyles = makeStyles({
+  root: {
+    ':hover': {
+      '& .fui-Checkbox__indicator': {
+        backgroundColor: tokens.colorCompoundBrandBackgroundHover,
+      },
+    },
+  },
+});
+```
+
+## Detailed Design or Proposal
+
+### Slot Names
+
+The root slot already has a static className with the format `fui-{ComponentName}`, per [#19937](https://github.com/microsoft/fluentui/issues/19937). This RFC proposes that every other slot of a component should have a static className added, with the format `fui-{ComponentName}__{slot}`. The static className would be added to the start before any other classNames.
+
+For example, the HTML tree of Checkbox would look like:
+
+```html
+<span className="fui-Checkbox ...other classes...">
+  <span className="fui-Checkbox__indicator ...other classes..."> ... </span>
+  <input className="fui-Checkbox__input ...other classes..." />
+  <label className="fui-Checkbox__label ...other classes..."> ... </label>
+</span>
+```
+
+### Exports
+
+These classNames could be exported from each components in an object. This would replace the existing `{component}ClassName` export.
```

Comment:
nit:

```suggestion
These classNames could be exported from each component in an object. This would replace the existing `{component}ClassName` export.
```